### PR TITLE
selection popup: edit role button

### PR DESCRIPTION
**Pull request by jakob223**
There's something funky going on here w/r/t the roles not showing their new values after the RPC goes through. I think the RPC needs to tell us the team changed, which it currently doesn't. But this was already more complicated than expected so maybe leaving it for another one.

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/teams/common/selection-popup.tsx [hunk 1]
```diff
@@ -309,7 +367,7 @@ const ChannelMembersActions = ({conversationIDKey, teamID}: ChannelActionsProps)
         onClick={onAddToChannel}
         fullWidth={Styles.isMobile}
       />
-      <Kb.Button label="Edit role" mode="Secondary" onClick={onEditRoles} fullWidth={Styles.isMobile} />
+      <EditRoleButton teamID={teamID} members={members} />
```
Comment (buoyad) [inline shared/teams/common/selection-popup.tsx]: I think if you've selected any owner when you're an admin we should remove or disable this, maybe with a tooltip.

File: shared/teams/common/selection-popup.tsx [hunk 2]
```diff
@@ -265,6 +266,61 @@ const TeamMembersActions = ({teamID}: TeamActionsProps) => {
...
+  }
+  const first = arr[0]
+  return arr.some(r => r !== first) ? null : first
+}
+const EditRoleButton = ({members, teamID}: {teamID: Types.TeamID; members: string[]}) => {
+  const dispatch = Container.useDispatch()
+
+  const teamDetails = Container.useSelector(state => Constants.getTeamDetails(state, teamID))
+  const roles = members.map(username => teamDetails.members.get(username)?.type)
+  const currentRole = allSameOrNull(roles)
```
Comment (buoyad) [inline shared/teams/common/selection-popup.tsx]: Makes sense for it to do this. I think we need to `setRole` to this when we show the popup or it's only set on mount.

File: shared/constants/teams.tsx [hunk 3]
```diff
@@ -422,13 +422,17 @@ export const getDisabledReasonsForRolePicker = (
       return subteamsCannotHaveOwners
     }
     if (yourRole !== 'owner') {
-      return onlyOwnersCanTurnTeamMembersInfoOwners
+      return onlyOwnersCanTurnTeamMembersIntoOwners
     }
-    const modifyingSelf = memberToModify === state.config.username
+    const modifyingSelf =
+      membersToModify === state.config.username ||
+      (typeof membersToModify === 'object' && membersToModify?.includes(state.config.username))
```
Comment (buoyad) [inline shared/constants/teams.tsx]: `Array.isArray(membersToModify)`

File: shared/teams/common/selection-popup.tsx (hunk 1 above)
Comment (jakob223) [reply shared/teams/common/selection-popup.tsx]: Nice, I like this. Disabled the button whenever there are no changes allowed.

Comment (buoyad) [reply shared/teams/common/selection-popup.tsx]: Cool. It doesn't show on tap on mobile. I think tooltip has a prop for that.

**Pull request merged**